data/tutorials/language/1ds_01_arrays.md

```diff
@@ -0,0 +1,158 @@
+
+```ocaml
+# Array.iter (fun x -> print_int x; print_string " ") zeroes;;
+1 2 42 4 5 - : unit = ()
+```
+
+### Map an Array
+
+The `Array.map` function applies a given function to each element of an array and returns a new array with the results. For example, to create a new array that contains the square of each element of the array `even_numbers` created above, we can apply the following square function `fun x -> x * x` to each element of the array:
+
+```ocaml
+# Array.map (fun x -> x * x) even_numbers;;
+- : int array = [|0; 4; 1764; 36; 64|]
+```
+
+### Folding an Array
+
+To combine all the elements of an array into a single result, we can use the `Array.fold_left` and `Array.fold_right` functions. These functions take a binary function, an initial accumulator value, and an array as arguments. The binary function takes two arguments: the current value of the accumulator and the current element of the array and returns a new accumulator value. Both functions traverse the array
+ but in opposite directions. This is essentially the same as `List.fold_left` and `List.fold_right`.
+
+Here is the signature of `Array.fold_left`:
+
+```ocaml
+val fold_left : ('a -> 'b -> 'a) -> 'a -> 'b array -> 'a
+```
+
+`fold_left f init a` computes `f (... (f(f init a.(0)) a.(1)) ...) a.(n-1)`
+
+Similarly, we can use the `Array.fold_right` function, which switches the order of its arguments:
+
+```ocaml
+val fold_right : ('b -> 'a -> 'a) -> 'b array -> 'a -> 'a
+```
+
+`fold_right f a init` computes `f a.(0) (f a.(1) ( ... (f a.(n-1) init) ...))`
+
+These functions derive a single value from the whole array. For example, they can be used to find the maximum element of an array:
+
+```ocaml
+# Array.fold_left max min_int even_numbers;;
+- : int = 42
+```
+
+Where `max` is the maximum function defined on elements of type int.
+
+### Sorting an Array
+
+To sort an array, we can use the `Array.sort` function. This function takes as arguments:
+- a comparison function
+- an array
+Ir sorts the provided array in place, in ascending order, according to the provided comparison function. Sorting performed by `Array.sort` modifies the content of the provided array, that why it returns `unit`. For example, to sort the array `even_numbers` created above, we can use:
+
+```ocaml
+# Array.sort compare even_numbers;;
+- : unit = ()
+# even_numbers;;
+- : int array = [|0; 2; 6; 8; 42|]
+```
+
+## Copying Part of an Array into Another Array
+
+The `Array.blit` function is available to efficiently copy a portion of an array into another array. Similarly to the `array.(x) <- y`  modification instruction, this function modifies the destination in place and does not return the modified array, but returns `unit`. Let’s suppose you want to copy a part of `even_numbers` into `zeroes`:
```

Comment:
```suggestion
The `Array.blit` function is available to efficiently copy a portion of an array into another array. Similar to the `array.(x) <- y` modification instruction, this function modifies the destination in place and does not return the modified array. Instead, it returns `unit`. Let’s suppose you want to copy a part of `even_numbers` into `zeroes`:
```

Reply:
```suggestion
The `Array.blit` function efficiently copies part of an array into an array. Similar to the `array.(x) <- y` operation, this function modifies the destination in place and returns `unit`, not the modified array. Suppose you wanted to copy a part of `even_numbers` into `zeroes`:
```
perhaps add an example of blitting an array onto itself